ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -105,9 +91,33 @@ void RSkComponentImage::OnPaint(
         canvas->restore();
     drawBorder(canvas,frame,imageBorderMetrics,imageProps.backgroundColor);
   } else {
-/* Emitting Image Load failed Event*/
-        imageEventEmitter->onError();
-    }
+  /* Emitting Image Load failed Event*/
+    RNS_LOG_INFO("image not loaded");
+    imageEventEmitter->onError();
+  }
+}
+
+sk_sp<SkImage> RSkComponentImage::getLocalImage(ImageSource source) {
+  std::string path;
+  if ( source.uri.empty() ) {
+    return nullptr;
+  }
+  if(source.uri.substr(0, 14) == "file://assets/") {
+    path = "./" + source.uri.substr(7);
+  }
+  RNS_PROFILE_START(getImageData)
+  sk_sp<SkImage> imageData = getImageData(path.c_str());
+  if(imageData) {
+    imageCacheManagerInstance_.imageDataInsertInCche(source.uri.c_str(), imageData);
+    RNS_LOG_INFO("New Entry in Map..."<<" file :"<<source.uri.c_str());
+  } else {
+    RNS_LOG_ERROR("Draw Image Failed :" << path);
+  }
+  RNS_PROFILE_END(path.c_str(),getImageData)
+  #ifdef RNS_IMAGE_CACHE_USAGE_DEBUG
+    printCacheUsage();
+  #endif //RNS_IMAGECACHING_DEBUG
+  return imageData;
```

Comment:
1. would suggest to club getLocalImage and getImageData into a single function, as its a single functionality.
2. Line 105-107 , can  be made as separate inline function.
     Because, in future if we have different local file folders, it will be easy to maintain the checks under a single function